deeployer.schema.json

```diff
@@ -10,19 +10,25 @@
 					"additionalProperties": false,
 					"properties": {
 						"replicas": {
-							"type": "string"
+							"type": "string",
+							"minLength": 1
 						},
 						"image": {
-							"type": "string"
+							"type": "string",
+							"minLength": 1
 						},
 						"ports": {
-							"type": "string"
+							"type": "array",
+							"minItems": 1
```

Comment:
There can be no ports in a container (so you should remove the "minItems" attribute here, it is ok if ports is an empty array)